test/helpers/integration.rb

```diff
@@ -139,4 +138,151 @@ def get_worker_pids(phase = 0, size = WORKERS)
     end
     pids.map(&:to_i)
   end
+
+  # used to define correct 'refused' errors
+  def thread_run_refused(unix: false)
+    if unix
+      [Errno::ENOENT, IOError]
+    else
+      DARWIN ? [Errno::ECONNREFUSED, Errno::EPIPE, EOFError] :
+        [Errno::ECONNREFUSED]
+    end
+  end
+
+  def cli_pumactl(argv, unix: false)
+    if unix
+      pumactl = IO.popen("#{BASE} bin/pumactl -C unix://#{@control_path} -T #{TOKEN} #{argv}", "r")
+    else
+      pumactl = IO.popen("#{BASE} bin/pumactl -C tcp://#{HOST}:#{@control_tcp_port} -T #{TOKEN} #{argv}", "r")
+    end
+    @ios_to_close << pumactl
+    Process.wait pumactl.pid
+    pumactl
+  end
+
+  def hot_restart_does_not_drop_connections(num_threads: 1, ttl_requests: 500)
```

Comment:
Just a nitpick: I read `ttl_requests` as time-to-live requests, which didn't really make sense.

`total_requests` would read a little easier.

Reply:
@cjlarose & @dentarg

Note to self - no more abbreviations in naming.  Fixed...